arkouda/strings.py

```diff
@@ -1071,6 +1073,13 @@ def endswith(self, substr: Union[bytes, str_scalars], regex: bool = False) -> pd
             substr = substr.decode()
         if not regex:
             substr = re.escape(substr)
+        if substr == "$":
+            # TODO remove once changes from chapel issue #20431 and #20441 are in arkouda
+            # no need to add "" case because it will be caught in contains
+            raise ValueError(
+                "regex operations not currently supported with a pattern=$ or pattern='' when "
+                "the empty string is contained in Strings"
+            )
```

Comment:
I know this condition is different, but I would like to see this incorporated to a sub-function as well. Maybe we can pass a `case` parameter to indicate which `if` to use?

Reply:
yeah this is actually the same case, it's just the `pattern=''` condition wasn't necessary, because it will be handled by the call to `self.contains(substr + "$", regex=True)`. So we can use the same helper across the board